### ISSUE TITLE
DPMSolverMultistepScheduler.step() in scorelidar/models/models.py --  p_sample_one_step

### ISSUE THREAD

**Issue by Fangjialema** [happyw1nd/ScoreLiDAR#8]
Hi~

There appears to be a issue in the sampling logic within p_sample_one_step, p_sample_one_step_val, and p_sample_loop methods. The code passes input_noise (defined as x_t - x_init) as the sample argument to DPMSolverMultistepScheduler.step(), which contradicts the standard expected input for this scheduler.

According to Hugging Face diffusers documentation, scheduler.step() should receive the current noisy sample x_t as the sample parameter (third arg). The scheduler uses this to compute the next denoised sample x_{t-1}.

Current Implementation in p_sample_one_step

``` python
# In p_sample_one_step, p_sample_one_step_val, and p_sample_loop
input_noise = x_t.F.reshape(t.shape[0],-1,3) - x_init  # (x_t - x_0 proxy)
x_t = x_init + self.dpm_scheduler.step(noise_t, t[0], input_noise)['prev_sample']
```
Here, input_noise (≈ x_t - x_0) is passed instead of x_t.

**Comment by happyw1nd** [comment]
Thank you for your interest in this project and for your sharp observation.

However, our approach to adding noise to x_0 differs from the conventional method. We use a technique we call "local noising," and the specific formula can be found in section 3.2 of the original paper.

Intuitively, at step t=T, this noising method is equivalent to adding unit Gaussian noise to the original coordinates of each point, rather than transforming the coordinates directly into a Gaussian distribution. The same principle applies to the denoising process. The original coordinates must be subtracted before denoising. You can conceptualize this as us denoising the "coordinate offset" rather than the absolute coordinates themselves.

Therefore, the operation x_t - x_init is performed to obtain this offset, which is the actual input for the scheduler.

**Comment by Fangjialema** [comment]
My bad  😂  That should be right.
